aiida/cmdline/commands/cmd_node.py

```diff
@@ -435,10 +435,11 @@ def verdi_graph():
     multiple=True
 )
 @click.option('-s', '--show', is_flag=True, help='Open the rendered result with the default application.')
+@click.option('--basename', default=None, help='Select file name')
```

Comment:
I wonder if it makes more sense to let the user simply decide the exact output filename and not just the basename. I don't really see a reason why the extension should always be the engine name. So maybe instead of this option, we allow the user to specify the output filename as an argument, which is a common design in CLIs. You can do this using one of our pre-defined arguments:
```suggestion
@arguments.OUTPUT_FILE(required=False)
```
You should change the `basename` argument in the `graph_generate` signature to `output_file`. If it is specified by the user, the value will be a [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html) instance.